### PR TITLE
pierre-ham-lint-apply-lint-fix-format

### PR DESCRIPTION
**Pull request by prenaux**
- [ ] Apply to all ham-test-thrift code
- [ ] Apply to ham's source code
- [ ] Apply to all ham/tests code
- [x] Apply to all ham/bin code
- [ ] Apply to all ham/specs/toolsets code
- [ ] Apply to all ham/toolsets code

### CODE REVIEW

**Comment by prenaux** [comment]
Current dependencies on/for this PR:
* master
  * **PR #32** <a href="https://app.graphite.dev/github/pr/prenaux/ham/32" target="_blank">Graphite</a> 
    * **PR #33** <a href="https://app.graphite.dev/github/pr/prenaux/ham/33" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/prenaux/ham/33?utm_source=stack-comment).

File: bin/ham-lint-fix-lib.sh [hunk 1]
```diff
@@ -5,22 +5,26 @@ function sh_lint() {
   toolset_import_once shell_linter >/dev/null
   errcheck $? "$SCRIPT_NAME" "Can't import shell_linter toolset." || return 1
 
+  local FILES=()
```
Comment (jamesrdd) [inline bin/ham-lint-fix-lib.sh:8]: Linting the lint lib itself, I love it.

File: _lint_all_files.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+sh bin/
```
Comment (jamesrdd) [inline _lint_all_files.txt:1]: Should we also lint/fix the top-level bash files in the repo? Although we would need a way to specify a linting directive to lint a directory non-recursively which I don't believe is possible right now.

**Review by jamesrdd — APPROVED** [review]
This shell linter is really nice, I look forward to standardizing all of our w3 scripts with it

File: _lint_all_files.txt (hunk 2 above)
Comment (prenaux) [reply _lint_all_files.txt:1]: Yes, but I'm leaning toward putting the file list a _build.ham file so we can pick/specify the files more precisely. This just bluntly goes everywhere-ish. That's problematic when there's 3rd party code and I feel that adding yet-another custom-ish pseudo glob language for linting won't buy us anything vs just doing it in a ham file...

**Comment by prenaux** [comment]
@prenaux started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/prenaux/ham/33).

**Comment by prenaux** [comment]
[Graphite](https://app.graphite.dev/github/pr/prenaux/ham/33) rebased this pull request as part of a merge.

**Comment by prenaux** [comment]
@prenaux merged this pull request with [Graphite](https://app.graphite.dev/github/pr/prenaux/ham/33).

**Pull request merged**